lib/cylc/cfgspec/globalcfg.py

```diff
@@ -275,17 +271,17 @@
     },
 
     'suite servers': {
-        'run hosts': vdr(vtype='string_list', default=["localhost"]),
+        'run hosts': vdr(vtype='string_list'),
```

Comment:
@matthewrmshin 

The removal of the `["localhost"]` default causes traceback if `[suite servers]run hosts` is not defined.

```
TypeError: object of type 'NoneType' has no len()
```

Why did we decentralise this?

Reply:
```
Traceback (most recent call last):
  File "mess", line 5, in <module>
    print ha.appoint_host()
  File "/home/h06/osanders/cylc/lib/cylc/scheduler_cli.py", line 444, in appoint_host
    self.hosts, ignore_if_thresholds_prov=True)
  File "/home/h06/osanders/cylc/lib/cylc/scheduler_cli.py", line 304, in _trivial_choice
    if self.selection_complete(host_list):
  File "/home/h06/osanders/cylc/lib/cylc/scheduler_cli.py", line 285, in selection_complete
    if len(host_list) == 0:
TypeError: object of type 'NoneType' has no len()
```

Reply:
This looks like it is because a null setting of the list here returns a `None` rather than an empty list. I'll add a local fix-and-test against this branch via a PR @sadielbartholomew.

(This default behaviour for settings expecting lists is a bit inconsistent, and can break on dump-and-load of a configuration file. Long term fix expected by #2609.)

Reply:
Thanks, @matthewrmshin. Note that there is now a new test (just merged in) which will confirm the behaviour is correct with ``[suite servers]run hosts`` not set (I didn't seem to pick up on unfortunately quite obvious this case when writing the original unit test). It currently fails on Travis CI, as well as for me locally.

Reply:
Should be addressed via sadielbartholomew/cylc#5.